src/actions/Follow.ts

```diff
@@ -0,0 +1,70 @@
+import { commands, DocumentLink, env, Uri, window } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+} from "../typings/Types";
+import displayPendingEditDecorations from "../util/editDisplayUtils";
+import { ensureSingleTarget } from "../util/targetUtils";
+
+export default class Follow implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const target = ensureSingleTarget(targets);
+
+    await displayPendingEditDecorations(
+      targets,
+      this.graph.editStyles.referenced
+    );
+
+    const link = await this.findLink(target);
+    if (link) {
+      await this.openUri(link.target!);
+    } else {
+      await this.graph.actions.executeCommand.run(
+        [targets],
+        "editor.action.revealDefinition",
+        { restoreSelection: false }
+      );
+    }
+
+    return {
+      thatMark: targets.map((target) => target.selection),
+    };
+  }
+
+  private async findLink(target: TypedSelection) {
+    const links = <DocumentLink[]>(
+      await commands.executeCommand(
+        "vscode.executeLinkProvider",
+        target.selection.editor.document.uri
+      )
+    );
```

Comment:
wow super annoying we need to get links for the entire document rather than just at a location.  Seems unnecessarily expensive if you're in a big document 🤔

Reply:
Oh yeah not saying it's your fault; just lamenting the api

Reply:
I didn't take it that way either. Just a bad api
I haven't actually tried to give it a second argument. Who knows it might actually work even though it's not in the documentation?

Reply:
Unfortunately, [no](https://github.com/microsoft/vscode/blob/54dec24a4d1e67dc6cb7fae0e6119dc84e0fab92/src/vs/editor/contrib/links/browser/getLinks.ts#L165) 😕

Reply:
Doesn't really surprise me